app/v1/controllers/validators/AmendPensionChargesRulesValidator.scala

```diff
@@ -0,0 +1,264 @@
+      overseasSchemeProviders.zipWithIndex.traverse_ { case (schemeProviderWithIndex, index) =>
+        import schemeProviderWithIndex._
+        pensionSchemeTaxReference
+          .traverse { references =>
+            validateReferences(s"$startOfPath/overseasSchemeProvider/$index", references)
+          }
+      }
+    }
+
+    def validateReferences(startOfPath: String, pensionSchemeTaxReference: Seq[String]): Validated[Seq[MtdError], Unit] = {
+      pensionSchemeTaxReference.zipWithIndex.traverse_ { case (reference, referenceIndex) =>
+        if (reference.matches(pensionSchemeTaxRefRegex)) {
+          valid
+        } else {
+          Invalid(List(PensionSchemeTaxRefFormatError.withPath(s"/$startOfPath/pensionSchemeTaxReference/$referenceIndex")))
+        }
+      }
+    }
+
+    combine(
+      pensionContributions
+        .traverse(pensionContributions => validateReferences("pensionContributions", pensionContributions.pensionSchemeTaxReference)),
+      pensionSavingsTaxCharges
+        .traverse(pensionSavingsTaxCharges => validateReferences("pensionSavingsTaxCharges", pensionSavingsTaxCharges.pensionSchemeTaxReference)),
+      pensionSchemeUnauthorisedPayments
+        .traverse(pensionSchemeUnauthorisedPayments =>
+          validateReferences("pensionSchemeUnauthorisedPayments", pensionSchemeUnauthorisedPayments.pensionSchemeTaxReference)),
+      pensionSchemeOverseasTransfers
+        .traverse(pensionSchemeOverseasTransfers =>
+          validatePensionSchemeTaxRef("pensionSchemeOverseasTransfers", pensionSchemeOverseasTransfers.overseasSchemeProvider)),
+      overseasPensionContributions
+        .traverse(overseasPensionContributions =>
+          validatePensionSchemeTaxRef("overseasPensionContributions", overseasPensionContributions.overseasSchemeProvider))
+    )
+  }
+
+  private def validateRuleIsAnnualAllowanceReduced(pensionSavingsTaxCharges: Option[PensionSavingsTaxCharges]): Validated[Seq[MtdError], Unit] = {
+    pensionSavingsTaxCharges
+      .map { taxCharges =>
+        import taxCharges._
+        isAnnualAllowanceReduced
+          .map { isAnnualAllowanceReduced =>
+            (isAnnualAllowanceReduced, taperedAnnualAllowance, moneyPurchasedAllowance) match {
+              case (false, _, _)         => valid
+              case (true, Some(true), _) => valid
+              case (true, _, Some(true)) => valid
+              case _                     => Invalid(List(RuleIsAnnualAllowanceReducedError))
+            }
+          }
+          .getOrElse(Invalid(List(RuleIncorrectOrEmptyBodyError)))
+      }
+      .getOrElse(valid)
+  }
+
+  private def validateCharges(pensionCharges: PensionCharges): Validated[Seq[MtdError], Unit] = {
+    import pensionCharges._
+
+    val fieldsWithPaths = List(
+      (
+        pensionSavingsTaxCharges.flatMap(_.benefitInExcessOfLifetimeAllowance.map(_.amount)),
+        s"/pensionSavingsTaxCharges/benefitInExcessOfLifetimeAllowance/amount"),
```

Comment:
lot's of strings starting with `s"..."` but no string interpolation being done so can be removed